src/device/wg_controller.rs

```diff
@@ -250,27 +246,52 @@ fn check_allowed_ips_correctness(peers: &BTreeMap<PublicKey, RequestedPeer>) ->
         .ok_or_else(|| Error::BadAllowedIps.into())
 }
 
-async fn consolidate_firewall<W: WireGuard, F: Firewall>(
+async fn consolidate_firewall<F: Firewall>(
     requested_state: &RequestedState,
-    wireguard_interface: &W,
     firewall: &F,
 ) -> Result {
-    let from_peers = wireguard_interface.get_interface().await?.peers;
-    let from_keys: HashSet<PublicKey> = from_peers.keys().copied().collect();
-    let to_peers: Vec<&telio_model::config::Peer> = requested_state
+    let from_keys_peer_whitelist: HashSet<PublicKey> =
+        firewall.get_peer_whitelist().iter().copied().collect();
+    let from_keys_ports_whitelist: HashSet<PublicKey> =
+        firewall.get_port_whitelist().keys().copied().collect();
+    let mut to_peers: HashMap<PublicKey, Node> = requested_state
         .meshnet_config
         .iter()
         .flat_map(|c| c.peers.iter())
         .flatten()
+        .map(|p| (p.public_key, p.into()))
         .collect();
-    let to_keys: HashSet<PublicKey> = to_peers.iter().map(|p| p.public_key).collect();
-    let delete_keys = &from_keys - &to_keys;
+    if let Some(exit_node) = &requested_state.exit_node {
+        if let Some(peer) = to_peers.get_mut(&exit_node.public_key) {
+            peer.is_exit = true;
```

Comment:
Just to double-check. This assignment will not affect the "real" `requested_state.meshnet_config` as `to_peers` is a copy of meshnet peer list?